pym/bob/scm/git.py

```diff
@@ -81,25 +91,61 @@ def getProperties(self):
                     ("refs/heads/" + self.__branch))
             )
         }]
+        for key, val in self.__remotes.items():
+            properties[0].update({GitScm.REMOTE_PREFIX+key : val})
+        return properties
 
     def asScript(self):
+        remotes_array = """
+# create an array of all remotes for this repository
+declare -A BOB_GIT_REMOTES=( [origin]="{URL}" )""".format(URL=self.__url)
+        if self.__remotes:
+            # add additional remotes to array
+            for name, url in self.__remotes.items():
+                remotes_array += """
+BOB_GIT_REMOTES[{NAME}]="{URL}" """.format(NAME=name, URL=url)
+        # create script to handle remotes
+        remotes_check_add_all = remotes_array + """
+# remove remotes from array that are already known to Git
+while read -r REMOTE_NAME ; do
+    # check for empty variable in case no remote at all is specified
+    if [ -z "$REMOTE_NAME" ]; then
+        continue
+    fi
+    # check if URL has changed
+    if [ ! "${BOB_GIT_REMOTES[$REMOTE_NAME]}" == "$(git remote get-url $REMOTE_NAME)" ]; then
+        git remote set-url "$REMOTE_NAME" "${BOB_GIT_REMOTES[$REMOTE_NAME]}"
```

Comment:
This should be moved below where it is clear that the remote is actually known to Bob. Otherwise the access to `"${BOB_GIT_REMOTES[$REMOTE_NAME]}"` will fail if the user added a remote manually.

Reply:
Moved, hopefully I got it right, this time.